src/lib.rs

```diff
@@ -29,7 +28,13 @@ impl SendDatagram for Room {
             client_infos: vec![],
         };
 
-        room_info_command.to_octets()
+        let mut stream = OutOctetStream::new();
+
+        room_info_command
+            .to_octets(&mut stream)
+            .expect("Failed to write command {room_info_command:?} to octet stream");
+
+        stream.data
```

Comment:
For symmetry it might be nicer if `SendDatagram::send` returns a opaque `impl WriteOctetStream` instead of the vector. However, that trait provides no way of getting a `Vec<u8>` so I did it like this for now.